packages/angular_devkit/build_angular/src/builders/browser-esbuild/options.ts

```diff
@@ -20,6 +20,20 @@ import { Schema as BrowserBuilderOptions, OutputHashing } from './schema';
 
 export type NormalizedBrowserOptions = Awaited<ReturnType<typeof normalizeOptions>>;
 
+/** Internal infrastructure settings hidden from builder schema but available when invoked programmatically. */
+export interface InfrastructureSettings {
+  /** Whether or not to write output files to disk. */
+  write?: boolean;
+
+  /**
```

Comment:
To follow up here, we discussed this somewhat in the design review today and generally agree that invoking the builder outside Architect is both required and undesirable as it produces a direct dependency on `browser-esbuild` specifically which should not be necessary. We decided that this is ok to start with and we should look into decoupling test builders from `browser-esbuild` later as we determine the right way of implementing them in Architect.

Reply:
Didn't we agree that since Architect is limiting we we should not using it and instead use the same pattern as the dev-server? 

https://github.com/angular/angular-cli/blob/866263b21e71d03800ae91033e951f9407ad0547/packages/angular_devkit/build_angular/src/builders/dev-server/vite-server.ts#L61

Nit: In general, I am not a big fan of adding not infra options such as entrypoint to the infrastructure options, but for now it's okay.  

I think where I'd like the builder to head is that there is a method which is exposed irrespective of the browser-builder options something along this lines. Where the latter would be invoked from "certain" builders that need custom logic not needed in browser builder. 

```ts
async function* buildEsbuildBrowser(
  userOptions: BrowserBuilderOptions,
  context: BuilderContext,
  infrastructureSettings?: {
    write?: boolean;
  },
): AsyncIterable<
  BuilderOutput & {
    outputFiles?: OutputFile[];
    assetFiles?: { source: string; destination: string }[];
  }
> {

  return build(userOptions, context, infrastructureSettings)
}

interface BuildOptions extends BrowserBuilderOptions {
  entryPoints: Record<string, string>,
  fileExtension: 'mjs'|'js',
  virtualModules: Record<string, string>,
}

export async function* buildEsbuildBrowser(
  options: NormalizedBrowserOptions,
  context: BuilderContext,
  infrastructureSettings?: {
    write?: boolean;
  },
): AsyncIterable<
  BuilderOutput & {
    outputFiles?: OutputFile[];
    assetFiles?: { source: string; destination: string }[];
  }
> {

  //

}
```

Reply:
Chatted offline to make sure I was understanding Alan's point here. The plan is to directly invoke ESBuild-er outside of Architect like Alan suggested. The concern here is that `InfrastructureSettings` is not really intended to represent internal-only options like I was using it for.

Instead, I refactored things to create a new `buildEsbuildBrowserInternal()` (I'm open to better naming suggestions) function as a separate entry point. This function also includes `entryPoints` and `outExtension` as internal options without putting them in `InfrastructureSettings` or requiring Architect to pass through that option. With that, I was able to revert all the Architect harness changes and simplify things there.

Hopefully this should address your concerns, but let me know if I'm still not quite understanding it.